File: app/src/main/java/com/example/android/effe2019/HomeAdapter.java
package com.example.android.effe2019;

import android.app.Activity;
import android.content.Intent;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.net.Uri;
import android.os.AsyncTask;
import android.text.Layout;
import android.util.Log;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.core.content.ContextCompat;

import java.io.InputStream;
import java.text.DateFormat;
import java.text.Format;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.List;

public class HomeAdapter extends ArrayAdapter {
    private Activity context;
    //list of users
    List<DataforHome> Users;
    public HomeAdapter(Activity context, List<DataforHome> Users) {
        super(context, R.layout.list_item_homefragment, Users);
        this.context = context;
        this.Users = Users;
    }



    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        if (convertView == null) {
            convertView = ((Activity) getContext()).getLayoutInflater().inflate(R.layout.list_item_homefragment, parent, false);
        }
        TextView title = (TextView) convertView.findViewById(R.id.updates);
        TextView date = (TextView) convertView.findViewById(R.id.date);
        TextView desc = (TextView) convertView.findViewById(R.id.desc);
//        View navigationView = convertView.findViewById(R.id.bottom_navigation_view_linear);
        final DataforHome task = (DataforHome) getItem(position);
        long timeStamp = task.getTimeStampofEvent();

        title.setText(task.getTitle());
        desc.setText(task.getDesciprionofEvent());
        date.setText(convertTime(timeStamp));
//        navigationView.setBackgroundColor(ContextCompat.getColor(this.context, R.color.red_bg_light));

        return convertView;
    }
    public String convertTime(long time){
        Date date = new Date(time);
        DateFormat dateFormat = new SimpleDateFormat("hh:mm a\ndd/MM");
        return dateFormat.format(date);
    }

}

File: app/src/main/java/com/example/android/effe2019/HomeAdapter.kt
package com.example.android.effe2019

import android.app.Activity
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView

import java.text.SimpleDateFormat
import java.util.Date

class HomeAdapter(
    private val context: Activity, //list of users
    internal var users: ArrayList<DataForHome>?
) : ArrayAdapter<DataForHome>(context, R.layout.list_item_homefragment,
    (users as MutableList<DataForHome>?)!!
) {


    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var convertView = convertView
        if (convertView == null) {
            convertView = (getContext() as Activity).layoutInflater.inflate(
                R.layout.list_item_homefragment,
                parent,
                false
            )
        }
        val title = convertView!!.findViewById<View>(R.id.updates) as TextView
        val date = convertView.findViewById<View>(R.id.date) as TextView
        val desc = convertView.findViewById<View>(R.id.desc) as TextView
        //        View navigationView = convertView.findViewById(R.id.bottom_navigation_view_linear);
        val task = getItem(position)
        val timeStamp = task!!.timestamp
        title.text = task.title
        desc.text = task.description
        date.text = convertTime(timeStamp!!)
        //        navigationView.setBackgroundColor(ContextCompat.getColor(this.context, R.color.red_bg_light));

        return convertView
    }

    private fun convertTime(time: Long): String {
        val date = Date(time)
        val dateFormat = SimpleDateFormat("hh:mm a\ndd/MM")
        return dateFormat.format(date)
    }

}
